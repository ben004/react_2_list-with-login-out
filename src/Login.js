import React, { Component } from "react";
import history from "./history";
import AddUser from "./Redux-storage/action";
import store from "./Redux-storage/store";

class login extends Component {
  handlClick() {
    let user = document.getElementById("name").value;
    let password = document.getElementById("pwd1").value;
    store.dispatch(AddUser({ user, password }));
    if (user === "" || password === "")
      document.getElementById("display").innerHTML = "*Fill the data for login";
    else history.push("/list");
  }
  render() {
    return (
      <div>
        {" "}
        <h1>Login page</h1>
        <br />
        Enter user name :
        <input
          type="text"
          data-testid="user_name"
          placeholder="Enter user name here"
          id="name"
        />
        <br />
        Enter password :
        <input
          type="password"
          data-testid="password"
          placeholder="Enter Password here"
          id="pwd1"
        />
        <br />
        <b>
          <p id="display" data-testid="para"></p>
        </b>
        <br />
        <button data-testid="login_button" onClick={this.handlClick}>
          login
        </button>
      </div>
    );
  }
}

export default login;
