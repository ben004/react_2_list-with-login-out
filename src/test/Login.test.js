import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import Login from "../Login";
import "@testing-library/jest-dom/extend-expect";
afterEach(cleanup);
const renderComponent = (props = {}) => render(<Login {...props} />);
describe("login", () => {
  it("should take a snapshot", () => {
    const tree = renderComponent();
    expect(tree).toMatchSnapshot();
  });
  it("should be enabled", () => {
    const { getByTestId } = renderComponent();
    expect(getByTestId("login_button").textContent).toBe("login");
  });
  it("should get user value", () => {
    const { getByTestId } = renderComponent();
    const btn = getByTestId("login_button");
    fireEvent.click(btn);
    expect(getByTestId("para").textContent).toBe("*Fill the data for login");
  });
  it("should get user value", () => {
    const { getByTestId } = renderComponent();
    const btn = getByTestId("login_button");
    const user = getByTestId("user_name");
    const password = getByTestId("password");
    fireEvent.change(user, { target: { value: "User@123 " } });
    fireEvent.change(password, { target: { value: "123456 " } });
    fireEvent.click(btn);
    expect(getByTestId("para").textContent).toBe("");
  });
  it("should get both user id and password", () => {
    const { getByTestId } = renderComponent();
    const btn = getByTestId("login_button");
    const user = getByTestId("user_name");
    fireEvent.change(user, { target: { value: "User@123" } });
    fireEvent.click(btn);
    expect(getByTestId("para").textContent).toBe("*Fill the data for login");
  });
});
