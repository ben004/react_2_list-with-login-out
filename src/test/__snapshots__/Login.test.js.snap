// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login should take a snapshot 1`] = `
<DocumentFragment>
  <div>
     
    <h1>
      Login page
    </h1>
    <br />
    Enter user name :
    <input
      data-testid="user_name"
      id="name"
      placeholder="Enter user name here"
      type="text"
    />
    <br />
    Enter password  :
    <input
      data-testid="password"
      id="pwd1"
      placeholder="Enter Password here"
      type="password"
    />
    <br />
    <b>
      <p
        data-testid="para"
        id="display"
      />
    </b>
    <br />
    <button
      data-testid="login_button"
    >
      login
    </button>
  </div>
</DocumentFragment>
`;
